crates/oxc_parser/src/diagnostics.rs

```diff
@@ -752,3 +752,10 @@ pub struct JSXExpressionsMayNotUseTheCommaOperator(
     #[label("JSX expressions may not use the comma operator. Did you mean to write an array?")]
     pub Span,
 );
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("Leading decorators must be attached to a class declaration")]
+#[diagnostic()]
+pub struct BadLeadingDecorator(
+    #[label("Leading decorators must be attached to a class declaration")] pub Span,
+);
```

Comment:
Currently we'll falsely accept this code:
```javascript
@a['b']
class A {}
```
Oxc silently eats `@a`, and parses `['b']` as a following ArrayExpression. Personally I'd like to make this part of early error pass. The decorator belongs to the ArrayExpression. And in the early error pass we identify this as illegal. 

But we could also do that directly in the parser.

Reply:
It's fine to leave it here for now. Only the ones that are on the hot path or have complicated logic are moved to the early pass.